### ISSUE TITLE
When updating datarule 'rule' it says it updates it but does not

### ISSUE THREAD

**Issue by maanenson** [exosite/exoline#116]
$ exo spec CIK spec.yaml --create
Running spec on: RID
updated rule for datarule_event_log_timeout
$ exo spec CIK spec.yaml
Running spec on: RID
spec expects rule for datarule_event_log_timeout to be:
{"timeout": {"repeat": false, "timeout": 3888000}}
...but it is:
{"timeout": {"repeat": false, "timeout": 6000}}

**Comment by dweaver** [comment]
I can't reproduce it. Are we on the same version?

```
$ exo --version
Exosite Command Line 0.9.16
$ CIK1=`curl cik.herokuapp.com`
$ exo info $CIK1 highTemp | jq .description.rule
{
  "timeout": {
    "repeat": false,
    "timeout": 3888000
  }
}
$ vi testspec.yaml
$ exo spec $CIK1 testspec.yaml --create
Running spec on: f84cee4c8ae88b72de4b512f63af05186d7e31b5
updated rule for highTemp
$ exo spec $CIK1 testspec.yaml
Running spec on: f84cee4c8ae88b72de4b512f63af05186d7e31b5
$ cat testspec.yaml
dataports:
    - alias: temp
      format: float

datarules:
    - alias: highTemp
      format: float
      subscribe: 21483ca91fab9c1c8b5f087c835a3a49cc6b7e6c
      rule: {"timeout": {"repeat": false, "timeout": 6000}}
$ exo info $CIK1 highTemp | jq .description.rule
{
  "timeout": {
    "repeat": false,
    "timeout": 6000
  }
}
```

**Comment by dweaver** [comment]
I see this issue only if I have retention set in the datarule.

```
$ cat repro.spec
dataports:
-   alias: device_event_log
    format: string
datarules:
-   alias: datarule_event_log_timeout
    format: integer
    retention:
        count: 10
        duration: infinity
    rule:
        timeout:
            repeat: false
            timeout: 3888000
    subscribe: device_event_log
$ CIK=`curl cik.herokuapp.com`
######################################################################## 100.0%
$ exo spec $CIK repro.spec --create
Running spec on: 43c5030ab69ef2e0f7bf08b2762a278a119546c6
device_event_log not found.
Creating dataport with name: device_event_log, alias: device_event_log, format: string
datarule_event_log_timeout not found.
Creating datarule with name: datarule_event_log_timeout, alias: datarule_event_log_timeout, format: integer, rule: {'timeout': {'repeat': False, 'timeout': 3888000}}
$ vi repro.spec
$ exo spec $CIK repro.spec --create
Running spec on: 43c5030ab69ef2e0f7bf08b2762a278a119546c6
updated rule for datarule_event_log_timeout
$ exo spec $CIK repro.spec
Running spec on: 43c5030ab69ef2e0f7bf08b2762a278a119546c6
spec expects rule for datarule_event_log_timeout to be:
{"timeout": {"repeat": false, "timeout": 6000}}
...but it is:
{"timeout": {"repeat": false, "timeout": 3888000}}
spec expects subscribe for datarule_event_log_timeout to be 5461692a907e4336c4856a99f934ce662d451928, but they are not.
```

**Comment by dweaver** [comment]
Fixed in 0.9.17
